provider/dynamodb.go

```diff
@@ -0,0 +1,301 @@
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	sn "github.com/mrz1836/go-sanitize"
+)
+
+type dynamodbTableKey struct {
+	Prefix, Feature, Variant string
+}
+
+func (t dynamodbTableKey) String() string {
+	marshalled, _ := json.Marshal(t)
+	return string(marshalled)
+}
+
+type dynamodbOnlineStore struct {
+	client *dynamodb.DynamoDB
+	prefix string
+	BaseProvider
+	timeout int
+}
+
+type dynamodbOnlineTable struct {
+	client    *dynamodb.DynamoDB
+	key       dynamodbTableKey
+	valueType ValueType
+}
+
+type dynamodbItem struct {
+	Entity string `dynamodbav:"Entity"`
+	Value  string `dynamodbav:"FeatureValue"`
+}
+
+type Metadata struct {
+	Tablename string `dynamodbav:"Tablename"`
+	Valuetype string `dynamodbav:"ValueType"`
+}
+
+func dynamodbOnlineStoreFactory(serialized SerializedConfig) (Provider, error) {
+	dynamodbConfig := &DynamodbConfig{}
+	if err := dynamodbConfig.Deserialize(serialized); err != nil {
+		return nil, err
+	}
+	if dynamodbConfig.Prefix == "" {
+		dynamodbConfig.Prefix = "Featureform_table__"
+	}
+	return NewDynamodbOnlineStore(dynamodbConfig)
+}
+
+func NewDynamodbOnlineStore(options *DynamodbConfig) (*dynamodbOnlineStore, error) {
+	config := &aws.Config{
+		Region:      aws.String(options.Region),
+		Credentials: credentials.NewStaticCredentials(options.AccessKey, options.SecretKey, ""),
+	}
+	sess := session.Must(session.NewSession(config))
+	dynamodbClient := dynamodb.New(sess)
+	if err := CreateMetadataTable(dynamodbClient); err != nil {
+		return nil, err
+	}
+	return &dynamodbOnlineStore{dynamodbClient, options.Prefix, BaseProvider{
+		ProviderType:   DynamoDBOnline,
+		ProviderConfig: options.Serialized(),
+	}, 10,
```

Comment:
This needs to be increased. Its only 10 seconds right now. Should be more like 2 minutes